### ISSUE TITLE
Mutating a SentinalArray element corrupts sentinel.

### ISSUE THREAD

**Issue by ablaom** [JuliaData/SentinelArrays.jl#25]
I admit I am new to this concept of sentinels. However, the following doesn't look right to me:

```julia
julia> a=SentinelArray(rand(2,3), NaN, missing)
2×3 SentinelArray{Float64,2,Float64,Missing,Array{Float64,2}}:
 0.248288  0.809637  0.191381
 0.697677  0.798701  0.542677

julia> a[1,1]=NaN
NaN

julia> a
2×3 SentinelArray{Float64,2,Float64,Missing,Array{Float64,2}}:
 NaN         0.809637  0.191381
   0.697677  0.798701  0.542677
```

I was expecting `missing` instead of `NaN` here.  The sentinel seems to have been corrupted:

julia> a.sentinel
7.645579181505823e54
```

**Comment by quinnj** [comment]
Here's what's going on:
-You created a `SentinelArray` with a sentinel of `NaN`; that means in the `Vector{Float64}` underneath, if there's a `NaN`, it will return `missing`
-You then _assigned_ the first index with `NaN`, which is the sentinel; internally, it doesn't want to corrupt data, so it chooses a new sentinel value automatically so that it doesn't clash with your data that now includes `NaN`

So basically, you should treat a `SentinelArray{T}` with `missing` as the return value exactly like you would treat `Vector{Union{T, Missing}}`; i.e. if you have a `missing` value, then you should do `a[1, 1] = missing`. SentinelArrays.jl will _actually_ set `NaN` under the hood, but it will return `missing` when you index that element. Hopefully that helps?

**Comment by ablaom** [comment]
Ah ha. I understand now. Thanks!
